File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/LineHeader.kt
package com.mbta.tid.mbta_app.android.component

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import com.mbta.tid.mbta_app.android.util.fromHex
import com.mbta.tid.mbta_app.model.Line
import com.mbta.tid.mbta_app.model.Route

@Composable
fun LineHeader(line: Line, routes: List<Route>, rightContent: (@Composable () -> Unit)? = null) {
    val route = routes.first()
    val (modeIcon, modeDescription) = routeIcon(route = route)
    TransitHeader(
        name = line.longName,
        backgroundColor = Color.fromHex(line.color),
        textColor = Color.fromHex(line.textColor),
        modeIcon = modeIcon,
        modeDescription = modeDescription,
        rightContent = rightContent
    )
}

File: iosApp/iosApp/ComponentViews/LineHeader.swift
//
//  LineHeader.swift
//  iosApp
//
//  Created by Simon, Emma on 6/18/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import Foundation
import Shared
import SwiftUI

struct LineHeader<Content: View>: View {
    let line: Line
    let routes: [Route]
    var rightContent: () -> Content?

    init(line: Line, routes: [Route], @ViewBuilder rightContent: @escaping () -> Content? = { EmptyView() }) {
        self.line = line
        self.routes = routes
        self.rightContent = rightContent
    }

    @ScaledMetric private var modeIconHeight: CGFloat = 24

    var body: some View {
        if let route = routes.first {
            TransitHeader(
                name: line.longName,
                routeType: route.type,
                backgroundColor: Color(hex: line.color),
                textColor: Color(hex: line.textColor),
                rightContent: rightContent
            )
            .accessibilityElement(children: .contain)
        } else {
            EmptyView()
        }
    }
}
